lex.go

```diff
@@ -43,14 +43,63 @@ var tokChanCap = TokenChanCap // capacity of lexer token channel
 func SetTokenChanCap(size int) {
 	tokChanCap = size
 }
-func Lex(reader io.Reader) chan NgxToken {
+
+// LexScanner provides an interface for scanning tokens.
+type LexScanner interface {
+	// Scan is used to advance the scanner to the next token.
+	Scan() bool
+	// Err returns any error encountered during scanning.
+	Err() error
+	// Text returns the most recent token generated by Scan as a string.
+	Text() string
+	// Line returns the line number of the most recent token generated by Scan.
+	Line() int
+}
+
+type ExtLexer interface {
+	Register(scanner LexScanner) []string
+	Lex(matchedToken string) <-chan NgxToken
+}
+
+type LexOptions struct {
+	ExternalLexers []ExtLexer
```

Comment:
Nit: A docstring here would help users understand what is being configured.

Reply:
done